### PR TITLE
pkg/labelsfilter: add more unit test and rewrite docs

### PR DESCRIPTION
**Pull request by aanm**
This change is only adding more unit tests to better understand the
behavior of the labelsfilter as well as improving the documentation for
the expectation of filtering labels.

Signed-off-by: André Martins <andre@cilium.io>

```release-note
Improve documentation of filtering unnecessary labels
```

### CODE REVIEW

File: pkg/labelsfilter/filter_test.go [hunk 1]
```diff
@@ -27,10 +25,6 @@ import (
 )
 
 // Hook up gocheck into the "go test" runner.
-func Test(t *testing.T) {
-	TestingT(t)
-}
-
```
Comment (tklauser) [inline pkg/labelsfilter/filter_test.go]: Why is this removed? Won't this lead to the `gocheck` tests no longer being run? I see no other place in this package where we hook up `gocheck` to `testing`.

File: Documentation/operations/scalability/identity-relevant-labels.rst [hunk 2]
```diff
@@ -89,10 +89,23 @@ will be used to evaluate Cilium identities:
...
-- app=*
-- name=*
+- io.kubernetes.pod.namespace*=.*
+- k8s-app*=*
+- app*=*
+- name*=*
+
+As well as, labels with the same prefix as defined in the configuration will
+also be considered, this lists some examples of labels that would also be
+evaluated for Cilium identities:
```
Comment (tklauser) [inline Documentation/operations/scalability/identity-relevant-labels.rst]: This sentence reads a bit strange. I think we can drop the "As well as" and make two sentences out of it:
```suggestion
Labels with the same prefix as defined in the configuration will
also be considered. This lists some examples of labels that would also be
evaluated for Cilium identities:
```

File: pkg/labelsfilter/filter_test.go (hunk 1 above)
Comment (aanm) [reply pkg/labelsfilter/filter_test.go]: This was not intended.

**Review by tklauser — APPROVED** [review]
